### PR TITLE
Fix message formatting for long truncated UTF8 strings

### PR DESCRIPTION
**Pull request by danog**
I'm also passing the intl error message to the exception to simplify debugging.

### CODE REVIEW

**Review by romm — CHANGES_REQUESTED** [review]
Thanks for the PR! Since I think this is not a common usecase, and that probably a lot of PHP apps are installed with the `mbstring` extension anyway, we'll use the `mb_strcut` function only when it is available.

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
     "require": {
         "php": "~8.0.0 || ~8.1.0 || ~8.2.0",
         "composer-runtime-api": "^2.0",
-        "psr/simple-cache": "^1.0 || ^2.0 || ^3.0"
+        "psr/simple-cache": "^1.0 || ^2.0 || ^3.0",
+        "symfony/polyfill-mbstring": "^1.27"
```
[the remark this answers is not in the document]
Comment (danog) [reply composer.json]: Reimplemented mb_strcut in pure PHP, instead :)

**Comment by danog** [comment]
Ping @romm? :)

**Comment by romm** [comment]
Hi @danog, sorry for the delay I've been quite busy lately.

I'm not against importing a polyfill in the library, but it would need unit tests. Have you been using parts of code from somewhere else?

Could you move the implementation in another class (a static method will be enough), and add unit tests? We'd need 100% Mutation Score Indicator on this (you can use `composer mutation` to check that).

Thanks for your work!

**Comment by danog** [comment]
Nah, wrote this myself just for this PR, that's OK, I imagined I'd need full mutation coverage, will do!

**Comment by danog** [comment]
@romm Should be good!

**Comment by danog** [comment]
Added you as a collaborator :)

**Comment by romm** [comment]
Thank you @danog, that was a long run but we made it! 😁

**Comment by danog** [comment]
Yay! :D 
Now for the general refactoring PR ;)

**Pull request merged**